Framework/Algorithms/src/Q1DWeighted.cpp

```diff
@@ -274,29 +480,25 @@ void Q1DWeighted::calculate(const MatrixWorkspace_const_sptr &inputWS) {
           m_normalisation[0][j][k] += w;
         }
 
-        if (m_nWedges != 0) {
-          // we do need to loop over all the wedges, since there is no
-          // restriction for those; they can also overlap
-          // that is the same pixel can simultaneously be in many wedges
-          for (size_t iw = 0; iw < m_nWedges; ++iw) {
-            double centerAngle =
-                static_cast<double>(iw) * M_PI / static_cast<double>(m_nWedges);
-            if (m_asymmWedges) {
-              centerAngle *= 2;
-            }
-            centerAngle += deg2rad * m_wedgeOffset;
-            const V3D subPix = V3D(position.X(), position.Y(), 0.0);
-            const double angle = fabs(
-                subPix.angle(V3D(cos(centerAngle), sin(centerAngle), 0.0)));
-            if (angle < deg2rad * m_wedgeAngle * 0.5 ||
-                (!m_asymmWedges &&
-                 fabs(M_PI - angle) < deg2rad * m_wedgeAngle * 0.5)) {
-              PARALLEL_CRITICAL(iqnorm_wedges) {
-                // first index 0 is the full azimuth, need to offset +1
-                m_intensities[iw + 1][j][k] += YIn[j] * w;
-                m_errors[iw + 1][j][k] += w * w * EIn[j] * EIn[j];
-                m_normalisation[iw + 1][j][k] += w;
-              }
+        for (size_t iw = 0; iw < m_nWedges; ++iw) {
+          double centerAngle = m_wedgesCenterAngle[iw];
+          const V3D subPix = V3D(position.X(), position.Y(), 0.0);
+          const V3D center = V3D(m_wedgesCenterX[iw], m_wedgesCenterY[iw], 0);
+          double angle =
+              fabs((subPix - center)
+                       .angle(V3D(cos(centerAngle), sin(centerAngle), 0.0)));
+
+          if ((angle < m_wedgesAngleRange[iw] * 0.5 ||
```

Comment:
This wasn't great, but is much worse now. Think of splitting to named booleans.

Reply:
Thinking of this a bit further, I think it would be neat to add a method like `checkPixelInSector` that will take a V3D and a struct Sector and return a boolean?